ict-weather-station/transformers/main/source.js

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ * 
+ *  Transformation code for ICT Weather Station Sensor 
+ * 
+ * ****************************************************************************/
+
+/**
+ * Extracts bytes from byte_payload from start to end and converts these bytes 
+ * into a float. Adapted from: 
+ * https://www.thethingsnetwork.org/forum/t/decode-float-sent-by-lopy-as-node/8757/2
+ * 
+ * @param {*} byte_payload payload in bytes to be converted to float.
+ * @param {*} start starting byte index in payload.
+ * @param {*} end ending byte index in payload.
+ */
+function extractFloat(byte_payload, start, end) {
+  // Extract bytes from payload, converting each byte into its unicode value
+  var bytes = [];
+  for(var i = start; i < end; i++) 
+    bytes.push(byte_payload[i].charCodeAt(0));
+
+  // Convert bytes to a float
+  var bits = bytes[0]<<24 | bytes[1]<<16 | bytes[2]<<8 | bytes[3];
+  var sign = (bits>>>31 === 0) ? 1.0 : -1.0;
+  var e = bits>>>23 & 0xff;
+  var m = (e === 0) ? (bits & 0x7fffff)<<1 : (bits & 0x7fffff) | 0x800000;
+  var float = sign * m * Math.pow(2, e - 150);
+
+  // Return float in 2 decimal places
+  return Math.round(float*100)/100;
```

Comment:
Don't mix presentation concerns i.e. rounding, here. Numbers can be rounded in the graphs etc.

Reply:
Complete!